### DISCUSSION TITLE
Provide prevMatch / nextMatch params to onEnter / onLeave

### DISCUSSION THREAD

**Discussion by ryanking1809** [TanStack/router#1695] [Ideas]
I have a project route with tabs which I use search params to navigate eg.
`project/$id?tab=tasks`
`project/$id?tab=notes`
`project/$id?tab=roles`

I want to warn the user if they leave without saving. However `onLeave` triggers on navigating between tabs (perhaps I've set something up incorrectly) and it only needed once a user leaves the project route.

Is it possible to add `nextMatch` to the params for `onLeave`? I currently import the router and it feels a bit off.
```js
import { router } from '../../router`
...
onLeave: (match) => {
        if (match.pathname !== router.latestLocation.pathname) {
            askToSave()
        }
    },
```

Perhaps this would be nicer as 
```js
onLeave: (currentMatch, nextMatch) => {
        if (currentMatch.pathname !== nextMatch.pathname) {
            askToSave()
        }
    },
```

**Comment by schiller-manuel** [comment]
shouldn't this functionality rather be implemented using `useBlocker`?

https://tanstack.com/router/v1/docs/framework/react/guide/navigation-blocking

**Reply by ryanking1809** [reply]
Yes, I think you're right. Thanks!
